### DISCUSSION TITLE
Can't get <pnp-popup> to work

### DISCUSSION THREAD

**Discussion by colonelclaypoo** [microsoft-search/pnp-modern-search#2894] [General]
Hello. A beginner's question if I may. I was looking to implement the

`<pnp-popup>
    <template id="popupclick">
        <span>Popup Example - Click Me</span>
    </template>
    <template id="popupheader">
        <span>Title: {{Title}}</span><br /><span>Header HTML here</span>
    </template>
    <template id="popupbody">
        <span>Title: {{Title}}</span><br /><span>Body HTML here</span>
    </template>
</pnp-popup>`

mentioned in the docs into the default slider template.
However, I can't figure out where to place it within <content id="data-content">

<div class="template">

    {{#if @root.properties.showSelectedFilters}}
        <pnp-selectedfilters 
            data-filters="{{JSONstringify filters.selectedFilters 2}}" 
            data-filters-configuration="{{JSONstringify filters.filtersConfiguration 2}}" 
            data-instance-id="{{filters.instanceId}}" 
            data-operator="{{filters.filterOperator}}" 
            data-theme-variant="{{JSONstringify @root.theme}}"
        >
        </pnp-selectedfilters>
    {{/if}}

    <div class="template--header">
        {{#if @root.properties.showResultsCount}}
            <div class="template--resultCount">
                <label class="ms-fontWeight-semibold">{{getCountMessage @root.data.totalItemsCount @root.inputQueryText}}</label>
            </div>
        {{/if}}

        <div class="template--sort">
            <pnp-sortfield 
                data-fields="{{JSONstringify @root.properties.dataSourceProperties.sortList}}" 
                data-default-selected-field="{{sort.selectedSortFieldName}}" 
                data-default-direction="{{sort.selectedSortDirection}}"
                data-theme-variant="{{JSONstringify @root.theme}}">
            </pnp-sortfield>    
        </div>
    </div>

    <pnp-slider data-items="{{JSONstringify @root.data.items 2}}" 
        data-options="{{JSONstringify @root.properties.layoutProperties.sliderOptions}}" 
        data-context="{{JSONstringify @root}}" 
        data-instance-id="{{@root.instanceId}}">

        <div class="template--slide" style="height: {{@root.properties.layoutProperties.slideHeight}}px;width: {{@root.properties.layoutProperties.slideWidth}}px;margin: 5px;">

                
                <pnp-documentcard 
                    data-ui-test-id="resultCard"	
                    data-title="\{{slot item @root.slots.Title}}"
                    data-location="\{{SiteTitle}}"
                    data-preview-image="\{{slot item @root.slots.PreviewImageUrl}}" 
                    data-preview-url="\{{slot item @root.slots.PreviewUrl}}" 
                    data-date="\{{getDate (slot item @root.slots.Date) 'LL'}}" 
                    data-href="\{{slot item @root.slots.Path}}" 
                    data-author="\{{slot item @root.slots.Author}}" 
                    data-profile-image="/_layouts/15/userphoto.aspx?size=L&username=\{{getUserEmail (slot item @root.slots.UserEmail)}}" 
                    data-file-extension="\{{slot item @root.slots.FileType}}"
                    data-is-container="\{{slot item @root.slots.IsFolder}}"
                    data-enable-preview="false" 
                    data-show-file-icon="true"
                    data-is-compact="false"
                    data-context="{{JSONstringify @root 2}}"
                    data-instance-id="{{@root.instanceId}}"
                    data-theme-variant="{{JSONstringify @root.theme}}"
                    data-allow-item-selection="{{@root.properties.itemSelectionProps.allowItemSelection}}"
                    data-item-key="{{@root.paging.currentPageNumber}}\{{@index}}"                     
                    data-selected-keys="{{JSONstringify @root.selectedKeys}}"
                    data-index="\{{@index}}"
                >
                </pnp-documentcard>
        </div>

    </pnp-slider>
</div>

Can you point me to where I have to put it?

Thanks.

**Comment by colonelclaypoo** [comment]
Perhaps it would help to to find an answer to a more generalized question. Where in the markup do I put the <pnp-> web components mentioned here? https://microsoft-search.github.io/pnp-modern-search/v3/search-parts/templating/#web-components
Thanks.

**Reply by wobba** [reply]
When you define a custom template you can add html/css into the edit box or reference an external file. This is where you would use the web component in the mark up.

**Comment by colonelclaypoo** [comment]
> <pnp-popup> <template id="popupclick"> <span>Popup Example - Click Me</span> </template> <template id="popupheader"> <span>Title: {{Title}}</span><br /><span>Header HTML here</span> </template> <template id="popupbody"> <span>Title: {{Title}}</span><br /><span>Body HTML here</span> </template> </pnp-popup>

Thanks Mikael. It seems wherever I place the web component...

`<pnp-popup> <template id="popupclick"> <span>Popup Example - Click Me</span> </template> <template id="popupheader"> <span>Title: {{Title}}</span><br /><span>Header HTML here</span> </template> <template id="popupbody"> <span>Title: {{Title}}</span><br /><span>Body HTML here</span> </template> </pnp-popup>`

...it won't take effect.

Other web components are placed in the `<div class="template"></>` such as `<pnp-filepreview>`. If I place `<pnp-popup>`here it doesn't seem to work. The pop-up example from the docs should work without any changes I would assume.

`<pnp-popup> <template id="popupclick"> <span>Popup Example - Click Me</span> </template> <template id="popupheader"> <span>Title: {{Title}}</span><br /><span>Header HTML here</span> </template> <template id="popupbody"> <span>Title: {{Title}}</span><br /><span>Body HTML here</span> </template> </pnp-popup>`

I hope you know what I mean. Thanks.

**Reply by wobba** [reply]
This is a v3 component and works just fine when I test it now.

Simple test template

```html
<content id="template">
    <pnp-popup>
    <template id="popupclick">
        <span>Popup Example - Click Me</span>
    </template>
    <template id="popupheader">
        <span>Title: {{Title}}</span><br /><span>Header HTML here</span>
    </template>
    <template id="popupbody">
        <span>Title: {{Title}}</span><br /><span>Body HTML here</span>
    </template>
    </pnp-popup>
</content>
```

**Reply by colonelclaypoo** [reply]
Thanks.
I guess I'm still doing something wrong.

This:
<content id="data-content">

    <style>

       /* Insert your CSS overrides here */
       .example-themePrimary a {
            color: {{@root.theme.palette.themePrimary}};
        }

        .icon {
            width: 20px;
            height: 16px;
        }

        ul.template--custom {
            list-style: none;
            padding-left: 5px;
        }

        ul.template--custom li {
            display: flex;
            padding: 8px;
        }


    </style>

    <content id="template">
    <pnp-popup>
    <template id="popupclick">
        <span>Popup Example - Click Me</span>
    </template>
    <template id="popupheader">
        <span>Title: {{Title}}</span><br /><span>Header HTML here</span>
    </template>
    <template id="popupbody">
        <span>Title: {{Title}}</span><br /><span>Body HTML here</span>
    </template>
    </pnp-popup>
</content>


<content id="placeholder-content">

    <style>

        .placeholder .icon {
            width: 20px;
            height: 16px;
            margin-right: 5px;
        }

        .placeholder ul {
            list-style: none;
        }

        .placeholder ul li {
            display: flex;
            align-items: center;
            padding: 8px;
        }

    </style>

    <div class="placeholder">
        <ul>
            {{#times @root.properties.paging.itemsCountPerPage}}
                <li>
                    <div class="icon placeholder--shimmer"></div>
                    <span class="placeholder--shimmer placeholder--line" style="width: 60%"></span>
                </li>
            {{/times}}
        </ul>
    </div>
</content>


...produces
![image](https://user-images.githubusercontent.com/29123261/232686994-8b086152-fdaf-4e3a-99b2-146c958066a9.png)

The only setting I changed after adding the results webpart is change the query template on page 1 of the settings.
Atems are returned if I chose any other template.

I'm on v4.8.0

Thanks for your help.

**Reply by wobba** [reply]
As I called out, it's a v3 feature - not migrated to v4.

**Reply by colonelclaypoo** [reply]
> As I called out, it's a v3 feature - not migrated to v4.

Oh, not migrated, now I get. Misinterpreted your statement before.
I see it working there now.

Thanks.
